lib/mllp/sender.ex

```diff
@@ -76,133 +68,91 @@ defmodule MLLP.Sender do
-          value |> String.to_charlist()
-      end
-
+    checked_address = normalize_address!(address)
     GenServer.start_link(__MODULE__, [address: checked_address, port: port] ++ options)
   end
 
-  @spec is_connected?(pid :: pid()) :: boolean()
-  def is_connected?(pid) do
-    GenServer.call(pid, :is_connected)
-  end
+  defp normalize_address!({_, _, _, _} = addr), do: addr
+  defp normalize_address!({_, _, _, _, _, _, _, _} = addr), do: addr
 
-  @spec reconnect(pid :: pid()) :: :ok
-  def reconnect(pid) do
-    GenServer.call(pid, :reconnect)
+  defp normalize_address!(addr) when is_binary(addr) do
+    addr
+    |> String.to_charlist()
+    |> parse_address!()
   end
 
-  @spec send_hl7_and_receive_ack(
-          pid :: pid,
-          payload :: HL7.Message.t(),
-          timeout :: non_neg_integer()
-        ) ::
-          MLLP.Ack.ack_verification_result() | {:error, SenderContract.sender_error()}
-  def send_hl7_and_receive_ack(
-        pid,
-        %HL7.Message{} = payload,
-        timeout \\ 5000
-      ) do
-    GenServer.call(pid, {:send_hl7_and_receive_ack, payload |> to_string()}, timeout + 1000)
+  defp normalize_address!(addr) when is_list(addr) do
+    parse_address!(addr)
   end
 
-  @spec send_hl7(pid :: pid, payload :: HL7.Message.t()) ::
-          {:ok, :sent} | {:error, SenderContract.sender_error()}
-  def send_hl7(pid, %HL7.Message{} = payload) when is_pid(pid) do
-    GenServer.call(pid, {:send_hl7, payload |> to_string()})
-  end
+  defp normalize_address!(addr), do: raise("Invalid server ip address : #{inspect(addr)}")
 
-  @spec send_non_hl7_and_receive_reply(
-          pid :: pid,
-          payload :: String.t(),
-          timeout :: non_neg_integer()
-        ) ::
-          {:ok, String.t()} | {:error, SenderContract.sender_error()}
-  def send_non_hl7_and_receive_reply(
-        pid,
-        payload,
-        timeout \\ 5000
-      ) do
-    options = [expect_reply: true, wrap_message: true, reply_timout: timeout]
+  defp parse_address!(addr) do
+    case :inet.parse_address(addr) do
+      {:error, _} ->
+        raise "Invalid server ip address : #{inspect(addr)}"
 
-    GenServer.call(pid, {:send, payload, options}, timeout + 1000)
+      {:ok, valid} ->
+        valid
+    end
   end
 
-  @spec send_non_hl7(pid :: pid, payload :: String.t()) ::
-          {:ok, :sent} | {:error, SenderContract.sender_error()}
-  def send_non_hl7(
-        pid,
-        payload
-      ) do
-    options = [expect_reply: false, wrap_message: true]
+  @spec is_connected?(pid :: pid()) :: boolean()
+  def is_connected?(pid), do: GenServer.call(pid, :is_connected)
 
-    GenServer.call(pid, {:send, payload, options})
-  end
+  @spec reconnect(pid :: pid()) :: :ok
+  def reconnect(pid), do: GenServer.call(pid, :reconnect)
 
-  @spec send_raw_and_receive_reply(
+  @default_send_opts %{
+    expect_reply: true,
+    timeout: 5000,
+    wrap_message: true,
+    reply_timeout: :infinity
+  }
+
+  @spec send(
```

Comment:
There's two things I don't like about this function name. 

1. It can be seen as redundant `Sender.send()` maybe `Client.send()` would read better
2. It sends AND receives, which is why I suspect it was `send_and_receive*` before yet, I don't like having an `and` in the function name either. OTOH, we can say, well sometimes sending implies a response and in the case of MLLP/HL7 that's always true. 

Suggestions and thoughts appreciated. What I do like about it, it's nice and short ;)

Reply:
I am ok with name `send`, but `Sender.send()` looks little weird and I like `Client.send()`. It's anyway is a MLLP client sending messages to a server.

Reply:
Let's rename it to Client then :) 👍